File: src/utils/chatbot.js
// src/chatbot.js
export const BOT_NAME = 'SimpleBot';
export const WELCOME_MESSAGE = 'Hello! How can I help you today?';

export function getWelcome() {
  return { bot_name: BOT_NAME, message: WELCOME_MESSAGE };
}

export function chat(userMessage) {
  // Simple echo logic
  return `${BOT_NAME}: You said '${userMessage}'`;
}

File: backend/chatbot.py
from fastapi import FastAPI, Request
from fastapi.responses import JSONResponse
from dotenv import load_dotenv
import os

load_dotenv("config.env")

app = FastAPI()

BOT_NAME = os.getenv("BOT_NAME", "Chatbot")
WELCOME_MESSAGE = os.getenv("WELCOME_MESSAGE", "Hello!")

@app.get("/api/welcome")
def welcome():
    return {"bot_name": BOT_NAME, "message": WELCOME_MESSAGE}

@app.post("/api/chat")
async def chat(request: Request):
    data = await request.json()
    user_message = data.get("message", "")
    # Simple echo logic
    response = f"{BOT_NAME}: You said '{user_message}'"
    return JSONResponse({"response": response})

# Moved to backend/
